src/config.ts

```diff
@@ -233,6 +236,9 @@ export class KubeConfig {
     }
 
     public addUser(user: User) {
+        if (!this.users) {
```

Comment:
Would it not be better to put this in the constructor? I think it would be more logical and easier to follow too.

Reply:
Not a blocker just curious before I add my lgtm (and it merges)

Reply:
The fields involved are public, so it's always possible that someone will overwrite them with something undefined or null, so the check here is safer.